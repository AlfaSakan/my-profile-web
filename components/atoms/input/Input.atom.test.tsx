import { fireEvent, render } from '@testing-library/react';
import renderer from 'react-test-renderer';
import Input from './Input.atom';

describe('input component test', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Input />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const tree = renderer.create(<Input value="test" />).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const onChange = jest.fn();
    const utils = render(<Input value={'val'} onChange={onChange} />);
    const input = utils.getByRole('textbox') as HTMLInputElement;

    fireEvent.change(input, { target: { value: 'change' } });
    expect(onChange).toHaveBeenCalled();
  });
});
