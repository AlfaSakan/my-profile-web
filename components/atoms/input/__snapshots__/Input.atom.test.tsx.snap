// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input component test should render correctly 1`] = `
<input
  className="w-full focus:outline-none rounded-md py-2 px-4 "
  placeholder="Type a message"
  type="text"
/>
`;

exports[`input component test should render correctly with props 1`] = `
<input
  className="w-full focus:outline-none rounded-md py-2 px-4 "
  placeholder="Type a message"
  type="text"
  value="test"
/>
`;
